src/Sarif.Sarifer/BackgroundAnalysisTextViewCreationListener.cs

```diff
@@ -51,6 +50,17 @@ public void TextViewCreated(ITextView textView)
                 {
                     this.sarifViewerInterop = new SarifViewerInterop(vsShell);
                 }
+
+                if (!this.sarifViewerInterop.IsSariferExtensionLoaded)
+                {
+                    // need to load package to get extension option setting values
+                    this.sarifViewerInterop.LoadSariferExtension();
+                }
+            }
+
+            if (!SariferOption.Instance.IsBackgroundAnalysisEnabled)
```

Comment:
>SariferOption.Instance.IsBackgroundAnalysisEnabled [](start = 17, length = 50)

This check should  be in all places that we analyze. this is something like "sarifer is enabled, if no, we should not trigger any kind of analysis"

Reply:
As discussed offline, this is option for automatic background analysis.

---
In reply to: [564952123](https://github.com/microsoft/sarif-visualstudio-extension/pull/354#discussion_r564952123) [](ancestors = 564952123)